### ISSUE TITLE
[Feature]: Configuration for manually recording pattern values

### ISSUE THREAD

**Issue by isvktt** [DeniedAccessLife/ArduinoStrike#7]
Could you create a config for us to manually set the recoil angles?
example :
ak 47 = 40, 20, 10, 120, 60
m4a1s = 5, 20, 50, 13, 19

**Comment by RedMemories** [comment]
if you check the code on c++ there are patterns, you just have to understand the coordinates as they go

**Comment by DeniedAccessLife** [comment]
Yes, that's right, changing the mouse movement coordinates is not a difficult action at all.

If you have a better option for the coordinates, you could share them, I would globally update these constants.

**Comment by isvktt** [comment]
I know there is a pattern
I'm saying put the weapons pattern in a config file
it would be very useful

**Comment by Kablonet** [comment]
Assigning the macro directly to the mouse1 button can cause gameplay issues in the game. For instance, the macro triggers even when throwing a flashbang. To avoid such issues, you can directly assign the macro to the mouse4 button. Alternatively, you can make the macro activate only when mouse1 is pressed while holding down the mouse4 button. I believe this will provide a more comfortable gaming experience.

**Comment by DeniedAccessLife** [comment]
> Assigning the macro directly to the mouse1 button can cause gameplay issues in the game. For instance, the macro triggers even when throwing a flashbang. To avoid such issues, you can directly assign the macro to the mouse4 button. Alternatively, you can make the macro activate only when mouse1 is pressed while holding down the mouse4 button. I believe this will provide a more comfortable gaming experience.

Yes, you correctly pointed out this situation. Personally, before throwing a grenade, I always disable the script using a hotkey. This might not be very convenient, as I have to constantly press keys and monitor the script’s status.

As for your suggestion, what if I use voice chat bound to the MOUSE5 or MOUSE4 button?

**Comment by DeniedAccessLife** [comment]
An update has been released, and the ability to install an optional key to confirm the operation of the recoil control has been added.

**Comment by Kablonet** [comment]
What exactly should we write for 'confirmation key (0/VK_CODE)?' If I had a choice, I would want the macro to be activated when I press and hold the mouse1 button while holding down the mouse4 button.

**Comment by DeniedAccessLife** [comment]
For your case:
MOUSE4 (XBUTTON1): 0x05
MOUSE5 (XBUTTON2): 0x06
The list of other constants is available on the Microsoft website.

**Comment by Kablonet** [comment]
The last update you made was very nice. Thank you. Additionally, I have one more request. I set the mouse sensitivity to 2 in the software settings. I also make the other necessary adjustments. However, when the weapon recoil control starts, there's a jitter at the crosshair on the screen. The weapon spray has become a bit harsh. If you could make this softer, it would be great. Sometimes the weapon spray behaves erratically, pulling the mouse down very quickly. The crosshair moves abruptly.

**Comment by DeniedAccessLife** [comment]
The sensitivity in the software is set relative to the game CS2, ranging from 1 to 8. Make sure the program's sensitivity matches the game's sensitivity to avoid abrupt movements during weapon recoil control. If the program's sensitivity differs from the game's sensitivity, it could cause unpredictable crosshair behavior.

**Comment by Kablonet** [comment]
It keeps giving the following error: Error checking available data: 6
Error checking available data: 6
Error checking available data: 6
Error checking available data: 6
Error checking available data: 6
Error checking available data: 6

**Comment by DeniedAccessLife** [comment]
How long have you been experiencing this error: 'Error checking available data: 6'? Did the code work for you before, or has this been a problem from the very beginning?

**Comment by DeniedAccessLife** [comment]
### 🌿 Closing due to inactivity, but feel free to reopen! 💖  

Hey there! 👋  

Since there hasn’t been any activity on this issue for a while, I’m closing it under the assumption that the problem has been resolved. If that's not the case, please don’t hesitate to reopen a new issue! 😊  

I'm always happy to help! 🚀
Thanks for your participation! ✨
